dissect/target/loaders/targetd.py

```diff
@@ -0,0 +1,135 @@
+    def _read(self, offset: int, length: int) -> bytes:
+        return b""
+
+
+class TargetdLoader(Loader):
+    instance = None
+
+    def __init__(self, path: Union[Path, str], **kwargs):
+        super().__init__(path)
+        self._plugin_func = None
+        self.client = None
+        self.output = None
+        self.peers = 0
+        uri = kwargs.get("parsed_path")
+        if uri is None:
+            raise LoaderError("No URI connection details have been passed.")
+        self.uri = uri.path
+        TargetdLoader.instance = self
+
+    @export(output="record", cache=False)
+    @arg(
+        "--host",
+        dest="host",
+        type=str,
+        default="localhost",
+        action="store",
+        help="IP-address of targetd broker",
+    )
+    @arg(
+        "--link",
+        dest="local_link",
+        type=str,
+        default="unix:///tmp/targetd",
+        action="store",
+        help="Domain socket or named pipe ('unix:///tmp/targetd')",
+    )
+    @arg(
+        "--port",
+        dest="port",
+        type=int,
+        default=1883,
+        action="store",
+        help="Port for connecting to targetd broker (1883)",
+    )
+    @arg(
+        "--adapter",
+        dest="adapter",
+        type=str,
+        default="Flow.Remoting",
+        action="store",
+        help="Adapter to use (default is 'Flow.Remoting')",
+    )
+    @arg(
+        "--peers",
+        dest="peers",
+        type=int,
+        action="store",
+        required=True,
+        help="Minimum number of hosts to wait for before executing query",
+    )
+    @arg("--help-targetd", action="help", help="Show help message for special targetd loader and exit")
+    @arg("-h", "--help", action="help", help="Show help message for plugin and exit")
+    def plugin_bridge(self, peers: int, host: str, local_link: str, port: int, adapter: str):
+        """Command Execution Bridge Plugin for Targetd.
+
+        This is a generic plugin interceptor that becomes active only if using
+        the targetd loader. This plugin acts as a bridge to connect to the Targetd broker
+        and will translate the requested plugin-operation into Targetd-commands using the selected
+        adapter (i.e. Flow.Remoting).
+        """
+
+        if not TARGETD_AVAILABLE:
+            raise ImportError("This loader requires the targetd package to be installed.")
+
+        self.output = None
+
+        if self.client is None:
+            self.client = Client(host, port, [self.uri], local_link, "targetd")
+            self.client.module_fullname = "dissect.target.loaders"
+            self.client.module_fromlist = ["command_runner"]
+            self.client.command = self._plugin_func
+            self.peers = peers
+            self.client.start()
+
+        while not self.output:
+            time.sleep(1)
+        return self.output
+
+    def get_command(self, func):
+        """For target API"""
+        self._plugin_func = func
```

Comment:
It feels a bit icky that this state separate form the `plugin_bridge` that is returned.
Would it be an idea to curry the `self.plugin_bridge` instead using `functools.partial()`?

Reply:
Hmm, that is a shame. I noticed there is also a `@functools.wraps`/`functools.update_wrapper()`, which turns a wrapper function into something that looks like the original. Maybe that can help?

Reply:
Probably even simpler:

```
curried_plugin_bridge = functools.update_wrapper(functools.partial(self.plugin_bridge, plugin_func=func), self.plugin_bridge)
```

Reply:
Interesting, giving this a try now...

Reply:
Okay this seems to work fine, but I still don't see how we can integrate this with the other 'static' part used for target-query https://github.com/fox-it/dissect.target/pull/188/files#diff-906948d654bf75fd862595a8e8df77d945b385ad7e2a898c8214e53b983a381dR128 ?

Reply:
Fixed